lib/lita/handlers/reviewme.rb

```diff
@@ -86,28 +88,19 @@ def generate_assignment(response, room: get_room(response))
         response.reply(reviewer.to_s)
       end
 
-      def comment_on_github(response, room: get_room(response))
-        repo = response.match_data[:repo]
-        id = response.match_data[:id]
-
-        reviewer = next_reviewer(room)
-        begin
-          pull_request = github_client.pull_request(repo, id)
-          owner = pull_request.user.login
-          reviewer = next_reviewer(room) if owner == reviewer
-        rescue Octokit::Error
-          response.reply("Unable to check who issued the pull request. Sorry if you end up being assigned your own PR!")
-        end
-        return response.reply('Sorry, no reviewers found') unless reviewer
-        comment = github_comment(reviewer, pull_request)
-
-        begin
-          github_client.add_comment(repo, id, comment)
-          response.reply("#{reviewer} should be on it...")
-        rescue Octokit::Error
-          url = response.match_data[:url]
-          response.reply("I couldn't post a comment. (Are the permissions right?) #{chat_mention(reviewer, url)}")
-        end
+      def review_on_github(response, room: get_room(response))
+        github = Lita::Reviewme::Github.new(config, response.match_data[:repo], response.match_data[:id])
+        reviewer = next_reviewer(room, github.owner)
+
+        github.assign_and_or_comment(reviewer)
+        response.reply("#{reviewer} should be on it...")
+      rescue Lita::Reviewme::NoReviewer
+        response.reply('Sorry, no reviewers found')
+      rescue Lita::Reviewme::UnknownOwner
```

Comment:
Probably no need to fully qualify these error names. If you take my advise about moving related errors into the GH class, it will have a nice readability to it, i.e. `rescue Github::UnknownOwner`.